spec/std/isa/inst/Zaamo/amoadd.SIZE.AQRL.layout

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+
+# yaml-language-server: $schema=../../../../schemas/inst_schema.json
+
+<%-
+  raise "'size' must be defined as 'b', 'h', 'w', or 'd'" unless %w[b h w d].include?(size)
+  raise "'aq' must be defined as true or false" unless [true, false].include?(aq)
+  raise "'rl' must be defined as true or false" unless [true, false].include?(rl)
+
+  size_info = {
+    "b" => { name: "byte", bits: 8, funct3: "000", operation_bits: "8" },
+    "h" => { name: "halfword", bits: 16, funct3: "001", operation_bits: "16" },
+    "w" => { name: "word", bits: 32, funct3: "010", operation_bits: "32" },
+    "d" => { name: "doubleword", bits: 64, funct3: "011", operation_bits: "64" }
+  }
+
+  current_size = size_info[size]
+
+  # Generate instruction name suffix based on aq/rl
+  aq_rl_suffix = ""
+  if aq && rl
+    aq_rl_suffix = ".aqrl"
+  elsif aq
+    aq_rl_suffix = ".aq"
+  elsif rl
+    aq_rl_suffix = ".rl"
+  end
+
+  # Generate match string with fixed aq/rl bits
+  aq_bit = aq ? "1" : "0"
+  rl_bit = rl ? "1" : "0"
+-%>
+
+$schema: "inst_schema.json#"
+kind: instruction
+name: amoadd.<%= size %><%= aq_rl_suffix %>
+long_name: Atomic fetch-and-add <%= current_size[:name] %><%= aq && rl ? " (acquire-release)" : (aq ? " (acquire)" : (rl ? " (release)" : "")) %>
+description: |
+  Atomically<%= aq && rl ? " with acquire and release ordering" : (aq ? " with acquire ordering" : (rl ? " with release ordering" : "")) %>:
+
+    * Load the <%= current_size[:name] %> at address _xs1_
+    * Write the <%= %w[b h].include?(size) ? "sign-extended value" : (size == "w" ? "sign-extended value" : "loaded value") %> into _xd_
```

Comment:
Why not combine "b", "h", and "w" here if they all emit the same string.
Does the behavior of "w" depend on XLEN, though, where it's only sign-extended if XLEN > 32?